File: java-spring/src/main/java/org/burgas/javaspring/entity/image/Image.java
package org.burgas.javaspring.entity.image;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.burgas.javaspring.entity.Entity;

import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@jakarta.persistence.Entity
@Table(name = "image", schema = "public")
public final class Image implements Entity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id")
    private UUID id;

    @Column(name = "name")
    private String name;

    @Column(name = "content_type")
    private String contentType;

    @JsonIgnore
    @Column(name = "data")
    private byte[] data;
}

File: kotlin-spring/src/main/kotlin/org/burgas/kotlinspring/entity/image/Image.kt
package org.burgas.kotlinspring.entity.image

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import org.burgas.kotlinspring.entity.Model
import java.util.UUID

@Entity
@Table(name = "image", schema = "public")
class Image : Model {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "id")
    override lateinit var id: UUID

    @Column(name = "name")
    lateinit var name: String

    @Column(name = "content_type")
    lateinit var contentType: String

    @JsonIgnore
    @Column(name = "data")
    lateinit var data: ByteArray
}